packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -566,6 +566,63 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     }).then<int>((_) => count);
   }
 
+  /// Triggers multiple frames for `duration` amount of time or until there are
+  /// no longer any frames scheduled, whichever is earlier.
+  ///
+  /// This does not promise settle of the widget, and there maybe remaining
+  /// animation and therefore undisposed `Ticker`s.
+  ///
+  /// This will call call [pump] at least once, even if no frames are scheduled
+  /// when the function is called, to flush any pending microtasks which may
+  /// themselves schedule a frame.
+  ///
+  /// `frequency` specify the frequency (in Hz) for frame requests. This is an
+  /// analog to screen refresh rate. The frequency rate will be rounded down to
+  /// integer microseconds frame interval. Default to be 59.94 Hz.
+  ///
+  /// This is useful when you are expecting there may be an infinite long
+  /// animations. To wait some finite animation you may want to use
+  /// [pumpAndSettle] instead.
+  ///
+  /// If the function returns, it returns the number of pumps that it performed.
+  /// When building and rendering frames is not expensive, the return value
+  /// should be `1 + (duration.inMicroseconds / (1E6 ~/ frequency)).floor()`
+  /// or `1 + (\[remaining animation time\] / (1E6 ~/ frequency)).ceil()`.
+  /// The formula may not be accurate when the division resul is very close to
+  /// an integer, due to fluctuation and/or truncation error.
+  ///
+  /// One can check if the return value from this function matches the expected
+  /// number of pumps to help cath regressions that cause loss of frame(s).
+  Future<int> pumpContinuous(Duration duration, {
+    EnginePhase phase = EnginePhase.sendSemanticsUpdate,
+    double frequency = 59.94,
+  }) async {
+    assert(duration != null);
+    assert(duration > Duration.zero);
+    assert(frequency > 0);
+    assert(() {
+      final WidgetsBinding binding = this.binding;
+      if (binding is LiveTestWidgetsFlutterBinding &&
+          binding.framePolicy == LiveTestWidgetsFlutterBindingFramePolicy.benchmark) {
+        throw 'When using LiveTestWidgetsFlutterBindingFramePolicy.benchmark, '
+              'hasScheduledFrame is never set to true. This means that pumpContinuous() '
+              'cannot be used, because it has no way to know if the application has '
+              'stopped registering new frames.';
+        }
+      return true;
+    }());
+    int count = 0;
+    final Duration interval = Duration(microseconds: 1E6 ~/ frequency);
+    return TestAsyncUtils.guard<void>(() async {
+      final DateTime endTime = binding.clock.fromNowBy(duration);
```

Comment:
If I understand correctly, this clock is a simulated clock that supports time travel (e.g., fast forward or slowdown). So we would render 60 frames for a 1-second-60-fps animation here no matter how slow the actual frame is. That is, if a frame takes 1 second to render, this could spend 1 minute rendering 60 frames and the simulated clock here think it's just 1 second.

Therefore, it may not necessarily simulate and benchmark how an app is performing on a real device. But it would still be very useful in terms of performance testing ( https://mail.google.com/mail/u/0/?pli=1#inbox/FMfcgxwHMZNWmbPBjLTWJhzlnLLfBplZ).

It would be nice if we could have both modes: (1) render only 1 frame for the 1-second-60-fps animation if a frame takes 1 second; (2) render 60 frames for 60 seconds if a frame takes 1 second for the 1-second-60-fps animation.

Reply:
https://api.flutter.dev/flutter/flutter_test/TestWidgetsFlutterBinding/clock.html I think for such purposes we should change the `clock` getter in `TestWidgetsFlutterBinding`, which is what `LiveTestWidgetsFlutterBinding` and `AutomatedTestWidgetsFlutterBinding` doing.

Reply:
I'm very unclear as to the value of this method in AutomatedTestWidgetsFlutterBinding. Why aren't we just throwing in that case?

This method seems like it is meant to measure the number of frames pumped for an animation on a real device with a real clock. Why would we want to measure any of that with fakeasync, which could get hung up by any number of problems in a test/benchmark such as trying to do real async work?

Reply:
To me, it seems the returned frame count isn't very interesting. I'll be more interested in the [FrameTiming](https://api.flutter.dev/flutter/dart-ui/FrameTiming-class.html) resulted from the frames driven by this.

From the meeting with Ming just a few minutes ago, I think `LiveTestWidgetsFlutterBinding ` may be less interesting to us in this case as we can just use `LiveTestWidgetsFlutterBindingFramePolicy.fullyLive` to drive the frames and completely remove the need of `pump` or `pumpContinuous`.

The interesting use case of `pumpContinuous` seems to be implementing a fake clock performance test suggested by Yegor in https://docs.google.com/document/d/1sGh6tvRUUNHI-VtnphliiYNJmmC_jo7afzQtAStPteg/edit?ts=5e80f082#bookmark=id.czojbloii9ww.

Reply:
> I'm very unclear as to the value of this method in AutomatedTestWidgetsFlutterBinding. Why aren't we just throwing in that case?
> 
> This method seems like it is meant to measure the number of frames pumped for an animation on a real device with a real clock. Why would we want to measure any of that with fakeasync, which could get hung up by any number of problems in a test/benchmark such as trying to do real async work?

The reason I return an int is to be consistent with `pumpAndSettle`.

Reply:
> To me, it seems the returned frame count isn't very interesting. I'll be more interested in the [FrameTiming](https://api.flutter.dev/flutter/dart-ui/FrameTiming-class.html) resulted from the frames driven by this.
> 
> From the meeting with Ming just a few minutes ago, I think `LiveTestWidgetsFlutterBinding ` may be less interesting to us in this case as we can just use `LiveTestWidgetsFlutterBindingFramePolicy.fullyLive` to drive the frames and completely remove the need of `pump` or `pumpContinuous`.
> 
> The interesting use case of `pumpContinuous` seems to be implementing a fake clock performance test suggested by Yegor in https://docs.google.com/document/d/1sGh6tvRUUNHI-VtnphliiYNJmmC_jo7afzQtAStPteg/edit?ts=5e80f082#bookmark=id.czojbloii9ww.

In my case `LiveTestWidgetsFlutterBindingFramePolicy.fullyLive` does work with an AnimationController-triggered animation, but from the wording I'm not sure if it's designed for synchronized running of the app, and if `pump` is completely not needed here (or at least can be replaced by Future.Delayed). I think I need confirm.

Reply:
I would find it very helpful to see a test that actually needs this as opposed to just using `fullyLive`. @liyuqian the linked doc isn't entirely clear to me, but I'm also not as familiar with that metric or its uses.  Also, FYI, the doc is google-only.

Reply:
Just not to be mixed with my other PRs / proposals, this was part of my plan but now it's more like a convenience wrap for infinite animation.